lib/matplotlib/axes/_axes.py

```diff
@@ -2104,7 +2117,37 @@ def step(self, x, y, *args, where='pre', data=None, **kwargs):
         -----
         .. [notes section required to get data note injection right]
         """
-        cbook._check_in_list(('pre', 'post', 'mid'), where=where)
+        cbook._check_in_list(('pre', 'post', 'mid', 'edges', 'between'),
+                             where=where)
+        cbook._check_in_list(('horizontal', 'vertical'),
+                             orientation=orientation)
+        if where == 'between' or where == 'edges':
+            if x.shape[0] != y.shape[0] + 1:
+                raise ValueError(f"When drawing with 'edges' or 'between', "
+                                 f"x must have first dimension greater "
+                                 f"then y by exactly 1, but x, y "
+                                 f"have shapes {x.shape} and "
+                                 f"{y.shape}")
+
+            if orientation == 'horizontal':
+                y = np.r_[y[0], y]
```

Comment:
Please see #16142 :wink:

Reply:
Alright., btw this PR is superseded by #15065, I should close it